### ISSUE TITLE
Variant is missing its product

### ISSUE THREAD

**Issue by Maximilian-Schwarz** [craftcms/commerce#1585]
**Description**
After the Upgrade from Craft Commerce 2 -> Craft Commerce 3 we are not able to delete a product.
Here we always got the error: "Variant is missing its product".

But if i look in the database and search with the right product id i can found all products in:
- commerce_products
- commerce_variants
- elements

Any ideas? I also try to disable the product and enable the product. But this also not working. I can't delete products.

**To Reproduce**
How, and what happened?
1. Go to the CP and click on "Commerce -> Products"
2. Select one product and click delete.
3. In the Ajax response you will see the error: "Variant is missing its product".

**Additional info**
- Craft version: 3.4.29.1
- Commerce version: 3.1.12
- PHP version: 7.2

**Comment by lukeholder** [comment]
Can you send your pre-updated DB and the Product ID affected to support@craftcms.com and we can take look. Thanks.

**Comment by Maximilian-Schwarz** [comment]
@lukeholder Hi Luke,

thanks for your reply. I found the issue. In the table commerce_variants here i found arouond 200 Variants where the productId was NULL. This variants does not exist. Now i have delete all variants with the productId NULL and everything is working.

I also found this migration "m180326_130001_cascadeDeleteVariants". But the migration is never running in my system. But now everything is working 👍
